lib/search.js

```diff
@@ -14,26 +14,21 @@ const PAGE = 1 // page_number is 1-based index
 // MAX_PAGE_ALLOWED = Math.ceil(MAX_RESULTS_POSSIBLE / PER_PAGE)
 const MAX_PAGE_ALLOWED = 34
 
-function search (q) {
-  return octokit.search
-    .issuesAndPullRequests(getSearchParams({ q }))
-    .then(({ data }) => {
-      if (data.total_count > data.items.length) {
-        const pageCount = Math.ceil(data.total_count / PER_PAGE)
-        const lastPageAllowed = pageCount > MAX_PAGE_ALLOWED ? MAX_PAGE_ALLOWED : pageCount
-        // page_number is 1-based index
-        var randomPage =
-          Math.floor(Math.random() * Math.floor(lastPageAllowed)) + 1
+async function search (q) {
+  const { data } = await octokit.search.issuesAndPullRequests(getSearchParams({ q }))
 
-        return octokit.search
-          .issuesAndPullRequests(getSearchParams({ q: q, page: randomPage }))
-          .then(({ data }) => {
-            return transform(data)
-          })
-      } else {
-        return transform(data)
-      }
-    })
+  if (data.total_count > data.items.length) {
+    const pageCount = Math.ceil(data.total_count / PER_PAGE)
+    const lastPageAllowed = pageCount > MAX_PAGE_ALLOWED ? MAX_PAGE_ALLOWED : pageCount
+    // page_number is 1-based index
+    const randomPage = Math.floor(Math.random() * Math.floor(lastPageAllowed)) + 1
+
+    return octokit.search
```

Comment:
Is it possible to convert this to async/await as well?
I believe this augments the readability to the changes proposed by you.

Reply:
Yep no problem 👍 Done in b3f4d15